### PR TITLE
RavenDB-7168

### PR DESCRIPTION
**Pull request by karmeli87**

### CODE REVIEW

File: src/Raven.Server/Documents/Indexes/IndexStore.cs [hunk 1]
```diff
@@ -343,7 +343,7 @@ public Task InitializeAsync(DatabaseRecord record)
         {
             if (_initialized)
                 throw new InvalidOperationException($"{nameof(IndexStore)} was already initialized.");
-
+        //    HandleDatabaseRecordChange();
```
Comment (ayende) [inline src/Raven.Server/Documents/Indexes/IndexStore.cs:346]: Why is this commented?

File: src/Raven.Server/Documents/Indexes/IndexStore.cs [hunk 2]
```diff
@@ -352,9 +352,9 @@ public Task InitializeAsync(DatabaseRecord record)
                 InitializePath(_documentDatabase.Configuration.Indexing.StoragePath);
 
                 _initialized = true;
-
-                return Task.Factory.StartNew(() => OpenIndexes(record), TaskCreationOptions.LongRunning);
             }
+
+            return Task.Factory.StartNew(HandleDatabaseRecordChange, TaskCreationOptions.LongRunning);
```
Comment (ayende) [inline src/Raven.Server/Documents/Indexes/IndexStore.cs:357]: Why are you running it like this?

File: src/Raven.Server/ServerWide/Maintance/ClusterObserver.cs [hunk 3]
```diff
@@ -69,7 +67,10 @@ class ClusterObserver : IDisposable
                     {
                         _logger.Info($"Closing observer on {_nodeTag}, caused by an interrupt.", e);
                     }
-                    return;
+                }
+                finally
+                {
+                    await delay;
```
Comment (ayende) [inline src/Raven.Server/ServerWide/Maintance/ClusterObserver.cs:73]: Should we _always_ wait, what about if there is an error?

Comment (karmeli87) [reply src/Raven.Server/ServerWide/Maintance/ClusterObserver.cs:73]: I think we should keep it alive, unless it is gracefully closed with the cancellation token.

File: src/Raven.Server/Documents/Indexes/IndexStore.cs (hunk 1 above)
Comment (karmeli87) [reply src/Raven.Server/Documents/Indexes/IndexStore.cs:346]: should be removed.

**Pull request merged**